lib/ansible/modules/cloud/amazon/aws_batch_compute_environment.py

```diff
@@ -0,0 +1,501 @@
+                    changed = True
+                    action_taken = "updated"
+                except (ParamValidationError, ClientError) as e:
+                    module.fail_json(msg="Unable to update environment: {0}".format(to_native(e)),
+                                     exception=traceback.format_exc())
+
+        else:
+            # Create Batch Compute Environment
+            changed = create_compute_environment(module, aws)
+            # Describe compute environment
+            action_taken = 'added'
+        response = get_current_compute_environment(module, aws)
+        if not response:
+            module.fail_json(msg='Unable to get compute environment information after creating')
+    else:
+        if current_state == 'present':
+            # remove the compute environment
+            changed = remove_compute_environment(module, aws)
+            action_taken = 'deleted'
+    return dict(changed=changed, batch_compute_environment_action=action_taken, response=response)
+
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   MAIN
+#
+# ---------------------------------------------------------------------------------------------------
+
+def main():
+    """
+    Main entry point.
+
+    :return dict: changed, batch_compute_environment_action, response
+    """
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(default='present', choices=['present', 'absent']),
+            compute_environment_name=dict(required=True),
+            type=dict(required=True, choices=['MANAGED', 'UNMANAGED']),
+            compute_environment_state=dict(required=False, default='ENABLED', choices=['ENABLED', 'DISABLED']),
+            service_role=dict(required=True),
+            compute_resource_type=dict(required=True, choices=['EC2', 'SPOT']),
+            minv_cpus=dict(type='int', required=True),
+            maxv_cpus=dict(type='int', required=True),
+            desiredv_cpus=dict(type='int'),
+            instance_types=dict(type='list', required=True),
+            image_id=dict(),
+            subnets=dict(type='list', required=True),
+            security_group_ids=dict(type='list', required=True),
+            ec2_key_pair=dict(),
+            instance_role=dict(required=True),
+            tags=dict(type='dict'),
+            bid_percentage=dict(type='int'),
+            spot_iam_fleet_role=dict(),
+            region=dict(aliases=['aws_region', 'ec2_region'])
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    # validate dependencies
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required for this module.')
+
+    aws = AWSConnection(module, ['batch'])
+
+    validate_params(module, aws)
+
+    results = manage_state(module, aws)
+
+    module.exit_json(**camel_dict_to_snake_dict(results))
```

Comment:
This will also snakify tags, so someone with a tag `MyThing` would actually see it changed to `my_thing` so we should use the `ignore` option on camel2snake to avoid messing with user tags.